finagle-core/src/main/scala/com/twitter/finagle/NameTreeParsers.scala

```diff
@@ -55,8 +55,17 @@ private class NameTreeParsers private (str: String) {
   }
 
   private[this] def eatWhitespace() {
-    while (!atEnd && str(idx).isWhitespace)
+    while (!atEnd && (str(idx).isWhitespace || str(idx) == '#')) {
+      if (str(idx) == '#') eatLine()
```

Comment:
As illustrated in the test case, /#foo names are safe. The comment character has to be found in the context of whitespace and not in a name. eatWhitespace isn't called while processing names.

Re: character, `;` is meaningful as a delimiter which Is problematic:
```
/goo => /car ; comment
          | /cab ;
```
Doesn't work.

I'm open to other suggestions but # seems like a natural comment char.